activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -63,8 +63,8 @@ def locking_enabled? #:nodoc:
 
         def increment_lock
           lock_col = self.class.locking_column
-          previous_lock_value = send(lock_col).to_i
-          send(lock_col + "=", previous_lock_value + 1)
+          previous_lock_value = _read_attribute(lock_col)
```

Comment:
It looks more consistent.
at least I think we need to remove redundant `.to_i` from this line, we expect that `lock_version` of type integer. http://edgeguides.rubyonrails.org/active_record_querying.html#optimistic-locking